### ISSUE TITLE
nucmer throws bad_alloc when reference is single line formatted

### ISSUE THREAD

**Issue by philippbayer** [mummer4/mummer#11]
I stumbled over this error:

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
ERROR: Failed to run nucmer, aborting.
```

with a large plant assembly and first thought that I was simply running out of memory, but it turns out the reference was formatted with one line for each pseudomolecule which caused the problem.

This is the reference in question: http://brassicadb.org/brad/datasets/pub/Genomes/Brassica_juncea/V1.5/Bju.genome.fasta.gz

My workaround is to use Biopython:

```python
from Bio import SeqIO
SeqIO.convert('Bju.genome.fasta' ,'fasta', 'Bju.genome_linewrapped.fasta', 'fasta')

```
which introduces proper line wrapping. I think it would be nicer if nucmer would at least complain about the missing line breaks, or introduce them itself during parsing.

**Comment by gmarcais** [comment]
Can you try the version in the `develop` branch? It should read a large input reference without problem.

**Comment by umngao** [comment]
I met the same error trying to compare two wheat chromosomes, each of size around 800 Mb using nucmer of the mummer-4.0.0beta1 build. Enlightened by this post, I also tried to change the line wrapping of my fasta files using Biopython. The error persisted. I tried to increase my machine memory to 300G, the problem remained the same. I further tried truncated versions of the two chromosomes ~70Mb each, I am waiting to see if that will work.
